docs/db-migration/adr/007-dual-read-migration.md

```
@@ -0,0 +1,176 @@
+    component_name  VARCHAR(255) PRIMARY KEY,
+    source          VARCHAR(10) NOT NULL DEFAULT 'git',  -- 'git' | 'db'
+    migrated_at     TIMESTAMP,
+    migrated_by     VARCHAR(255)
+);
+```
+
+- On initial deployment: all existing components have `source = 'git'`
+- New components created via API/UI: inserted with `source = 'db'`
+- After successful import + validation: updated to `source = 'db'`
+
+### Routing Resolver
```

markdownlint-cli2 (0.21.0)

[warning] Headings should be surrounded by blank lines
Expected: 1; Actual: 0; Below

(MD022, blanks-around-headings)